charts/partials/templates/_affinity.tpl

```diff
@@ -1,38 +1,11 @@
-{{ define "linkerd.pod-affinity" -}}
-podAntiAffinity:
-  preferredDuringSchedulingIgnoredDuringExecution:
-  - podAffinityTerm:
-      labelSelector:
-        matchExpressions:
-        - key: {{ default "linkerd.io/control-plane-component" .label }}
-          operator: In
-          values:
-          - {{ .component }}
-      topologyKey: failure-domain.beta.kubernetes.io/zone
-    weight: 100
-  requiredDuringSchedulingIgnoredDuringExecution:
-  - labelSelector:
-      matchExpressions:
-      - key: {{ default "linkerd.io/control-plane-component" .label }}
-        operator: In
-        values:
-        - {{ .component }}
-    topologyKey: kubernetes.io/hostname
-{{- end }}
-
 {{ define "linkerd.node-affinity" -}}
 nodeAffinity:
 {{- toYaml .Values.nodeAffinity | trim | nindent 2 }}
 {{- end }}
 
 {{ define "linkerd.affinity" -}}
```

Comment:
Since node-affinity is now the only thing in this define, I think we can just inline the node-affinity define here to simplify.